crates/holochain_p2p/src/types/event.rs

```diff
@@ -110,225 +108,96 @@ pub enum CountersigningSessionNegotiationMessage {
-        match $h {
-            HolochainP2pEvent::CallRemote { $i, .. } => { $($t)* }
-            HolochainP2pEvent::Get { $i, .. } => { $($t)* }
-            HolochainP2pEvent::GetMeta { $i, .. } => { $($t)* }
-            HolochainP2pEvent::GetLinks { $i, .. } => { $($t)* }
-            HolochainP2pEvent::CountLinks { $i, .. } => { $($t)* }
-            HolochainP2pEvent::GetAgentActivity { $i, .. } => { $($t)* }
-            HolochainP2pEvent::MustGetAgentActivity { $i, .. } => { $($t)* }
-            HolochainP2pEvent::ValidationReceiptsReceived { $i, .. } => { $($t)* }
-            HolochainP2pEvent::SignNetworkData { $i, .. } => { $($t)* }
-            HolochainP2pEvent::CountersigningSessionNegotiation { $i, .. } => { $($t)* }
-            $($t2)*
-        }
-    };
-}
-
-impl HolochainP2pEvent {
-    /// The dna_hash associated with this network p2p event.
-    pub fn dna_hash(&self) -> &DnaHash {
-        match_p2p_evt!(self => |dna_hash| { dna_hash }, {
-            HolochainP2pEvent::Publish { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::FetchOpData { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::QueryOpHashes { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::QueryAgentInfoSigned { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::QueryAgentInfoSignedNearBasis { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::QueryGossipAgents { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::PutAgentInfoSigned { dna_hash, .. } => { dna_hash }
-            HolochainP2pEvent::QueryPeerDensity { dna_hash, .. } => { dna_hash }
-        })
-    }
-
-    /// The agent_pub_key associated with this network p2p event.
-    pub fn target_agents(&self) -> &AgentPubKey {
-        match_p2p_evt!(self => |to_agent| { to_agent }, {
-            HolochainP2pEvent::Publish { .. } => { unimplemented!("There is no single agent target for Publish") }
-            HolochainP2pEvent::FetchOpData { .. } => { unimplemented!("There is no single agent target for FetchOpData") }
-            HolochainP2pEvent::QueryOpHashes { .. } => { unimplemented!("There is no single agent target for QueryOpHashes") }
-            HolochainP2pEvent::QueryAgentInfoSigned { .. } => { unimplemented!("There is no single agent target for QueryAgentInfoSigned") },
-            HolochainP2pEvent::QueryAgentInfoSignedNearBasis { .. } => { unimplemented!("There is no single agent target for QueryAgentInfoSignedNearBasis") },
-            HolochainP2pEvent::QueryGossipAgents { .. } => { unimplemented!("There is no single agent target for QueryGossipAgents") },
-            HolochainP2pEvent::PutAgentInfoSigned { .. } => { unimplemented!("There is no single agent target for PutAgentInfoSigned") },
-            HolochainP2pEvent::QueryPeerDensity { .. } => { unimplemented!() },
-        })
-    }
+/// Handle requests made by remote peers.
+pub trait HcP2pHandler: 'static + Send + Sync + std::fmt::Debug {
```

Comment:
This file used to be the ghost_actor event channel for legacy kitsune to make requests into holochain.

Now it is a simple kitsune2-style trait that can be type-erased into a dynamic trait-object.